user-console/src/main/resources/org/pentaho/mantle/public/browser/js/browser.js

```diff
@@ -1131,8 +1125,54 @@ define([
       var $target = $(event.currentTarget).parent().parent();
       if ($target.hasClass("open")) {
         $target.removeClass("open").find("> .folders").hide();
-      } else {
+      } else if ($target.find("> .folders").children().length > 0) {
         $target.addClass("open").find("> .folders").show();
+      } else {
+        var path = $target.attr("path");
+        var myself = this;
+
+        var url = CONTEXT_PATH + "api/repo/files/" +
+            FileBrowser.encodePathComponents(path == null ? ":" : Encoder.encodeRepositoryPath(path))
+            + "/tree?depth=1&showHidden=" + myself.model.get("showHiddenFiles") + "&filter=*%7CFOLDERS";
+        $.ajax({
```

Comment:
Not high priority. But this file is huge... is there any way this can be in a different class? (The object being past into the ajax)

Reply:
@cseverino789 It is a huge file, with many ajax calls. While your notion is definitely correct, that would require changing all 7 of the ajax calls to utilize a new class. I'm up for it if we deem it necessary.